Userland/Libraries/LibGfx/PortableImageMapLoader.h

```diff
@@ -86,9 +93,11 @@ IntSize PortableImageDecoderPlugin<TContext>::size()
         return {};
 
     if (m_context->state < TContext::State::Decoded) {
-        bool success = decode(*m_context);
-        if (!success)
+        if (decode(*m_context).is_error()) {
+            m_context->state = TContext::State::Error;
+            // FIXME: We should propagate errors
```

Comment:
(not sure if this api is good enough; by calling `size()` iʼd like to have it just accessed and provided—decoding should happen elsewhere and before i can even call this method… not sure how to structure that well enough though, maybe by `decode` returning optionally an object that has this method to be called? 🤔 [this would probably be a builder pattern around state changes which yields a usable object to work on after decoding…?])

Reply:
This PR is not about changing the API of `ImageDecoderPlugin`, feel free to open a PR to enhance it.